src/include/utils/elog.h

```diff
@@ -504,6 +504,7 @@ extern ErrorData *CopyErrorData(void);
 extern void FreeErrorData(ErrorData *edata);
 extern void FlushErrorState(void);
 extern void ReThrowError(ErrorData *edata)  __attribute__((__noreturn__));
+extern void ThrowErrorData(ErrorData *edata);
```

Comment:
```suggestion
extern void ThrowErrorData(ErrorData *edata) __attribute__((__noreturn__));
```

Seems that we need a no-return attribute here?

Reply:
There is a macro `pg_attribute_noreturn()`.

Reply:
hi @higuoxing  and @Aegeaner  here we can use __attribute__((__noreturn__)
because there is a return statement in ThrowErrorData.
if we use  __attribute__((__noreturn__) when declare ThrowErrorData, there is a warning.
```
elog.c:1972:3: warning: function declared ‘noreturn’ has a ‘return’ statement
   return;     /* error is not to be reported at all */
```

```
void
ThrowErrorData(ErrorData *edata)
{
	ErrorData  *newedata;
	MemoryContext oldcontext;

	if (!errstart(edata->elevel, edata->filename, edata->lineno, edata->funcname, TEXTDOMAIN))
		**return;**	
....
}